tests/conftest.py

```diff
@@ -203,34 +203,37 @@ def event_loop() -> asyncio.AbstractEventLoop:  # type: ignore
 
 
 @pytest.fixture(scope="session")
-def client() -> SimpleCacheClient:
+def client() -> Iterator[SimpleCacheClient]:
     configuration = Laptop.latest()
     credential_provider = CredentialProvider.from_environment_variable("TEST_AUTH_TOKEN")
     with SimpleCacheClient(configuration, credential_provider, DEFAULT_TTL_SECONDS) as _client:
         # Ensure test cache exists
-        _client.create_cache(TEST_CACHE_NAME)
+        _client.create_cache(cast(str, TEST_CACHE_NAME))
         try:
             yield _client
         finally:
-            _client.delete_cache(TEST_CACHE_NAME)
+            _client.delete_cache(cast(str, TEST_CACHE_NAME))
 
 
 @pytest_asyncio.fixture(scope="session")
-async def client_async() -> SimpleCacheClientAsync:
+async def client_async() -> AsyncIterator[SimpleCacheClientAsync]:
     configuration = Laptop.latest()
     credential_provider = CredentialProvider.from_environment_variable("TEST_AUTH_TOKEN")
     async with SimpleCacheClientAsync(configuration, credential_provider, DEFAULT_TTL_SECONDS) as _client:
         # Ensure test cache exists
         # TODO consider deleting cache on when test runner shuts down
-        await _client.create_cache(TEST_CACHE_NAME)
+        await _client.create_cache(cast(str, TEST_CACHE_NAME))
         try:
             yield _client
         finally:
-            await _client.delete_cache(TEST_CACHE_NAME)
+            await _client.delete_cache(cast(str, TEST_CACHE_NAME))
+
+
+TUniqueCacheName = Callable[[SimpleCacheClient], str]
 
 
 @pytest.fixture
-def unique_cache_name(client: SimpleCacheClient) -> str:
+def unique_cache_name(client: SimpleCacheClient) -> Iterator[Callable[[SimpleCacheClient], str]]:
```

Comment:
same question here about iterator

Reply:
Same thing: because the fixture has setup and teardown, we use a yield statement (the setup goes before and teardown after). `Iterator` the preferred annotation for this; tests that consume the fixture only need to write `unique_cache_name: TUniqueCacheName` (see type alias just above this)

Reply:
Same as the other fixture: because the fixture has setup and teardown, we use a yield statement (the setup goes before and teardown after). `Iterator` is the preferred annotation for this; tests that consume the fixture only need to write `unique_cache_name: TUniqueCacheName` (see type alias just above this)

Reply:
For reference: https://docs.pytest.org/en/6.2.x/fixture.html